ns1/resource_zone.go

```diff
@@ -279,11 +288,18 @@ func resourceDataToZone(z *dns.Zone, d *schema.ResourceData) {
 // zoneCreate creates the given zone in ns1
 func zoneCreate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*ns1.Client)
-	z := dns.NewZone(d.Get("zone").(string))
+	zone := d.Get("zone").(string)
+	z := dns.NewZone(zone)
```

Comment:
this seems like the right approach, but my delicate sensibilities prefer leaving this line alone, and pulling the zoneName back out of `z` in the `if` clause.
not a very strong opinion and not based on anything i've measured, so if consensus is to keep it, i'd only ask we call it `zoneName` instead of `zone`